Source/Optional.swift

```diff
@@ -1,20 +1,70 @@
+/**
+    map a function over an optional value
+
+    - If the value is .None, the function will not be evaluated and this will return .None
+    - If the value is .Some, the function will be applied to the unwrapped value
```

Comment:
do you want to use mapped here instead of applied to keep with the function name?